src/App.tsx

```diff
@@ -4,10 +4,26 @@ import Carousel from './components/Carousel';
 
 interface State {
   images: string[];
+  step: string
+  itemWidth: string
+  frameSize: string
+  animationDuration: string
+  infinite: boolean
+  scrollFn: (arg: number) => void
+  scroll: number
 }
 
 class App extends React.Component<{}, State> {
-  state = {
+  state: Readonly<{
+    images: string[]
+    step: string
+    itemWidth: string
+    frameSize: string
+    animationDuration: string
+    infinite: boolean
+    scroll: number
+    scrollFn: (arg: number) => void
```

Comment:
```suggestion
    scrollFn: (arg: number) => void
```
why do you need this function in the state?